File: Internship Tracking System - Company/src/DBOperations.cs
using MySql.Data.MySqlClient;
using System;
using System.Collections.Generic;
using System.Configuration;
using System.Data;
using System.Data.SqlClient;
using System.Linq;
using System.Web;
using System.Web.UI;

namespace Internship_Tracking_System___Company
{

    public class DBOperations
    {
        public static String connString = ConfigurationManager.ConnectionStrings["MySQL"].ToString();
        public static string CompanyID = "";
        public static string companyMail = "";
        public static int internshipBookID;

        public static bool loginCheck(int companyID, string companyPassword)
        {
            MySqlConnection conn = new MySqlConnection(connString);

            string sql = "SELECT * FROM company WHERE companyID=@pcID and companyPassword=@pcPassword";

            MySqlCommand cmd = new MySqlCommand(sql, conn);

            cmd.Parameters.AddWithValue("@pcID", companyID);
            cmd.Parameters.AddWithValue("@pcPassword", companyPassword);

            MySqlDataAdapter adaptor = new MySqlDataAdapter();
            adaptor.SelectCommand = cmd;

            DataSet dS = new DataSet();

            conn.Open();
            adaptor.Fill(dS);
            conn.Close();

            try
            {
                CompanyID = dS.Tables[0].Rows[0][0].ToString();
            }

            catch (IndexOutOfRangeException)
            {

            }

            bool result = false;

            if (dS.Tables[0].Rows.Count > 0)
            {
                result = true;
            }
            else
            {
                result = false;
            }

            return result;
        }

        public static void setInternshipBookConfrim(string temp)
        {
            MySqlConnection conn = new MySqlConnection(connString);

            string sql = "update internshipbook set companyBookCheck=@psCBC where internshipBookID=@psIBID";

            MySqlCommand cmd = new MySqlCommand(sql, conn);

            cmd.Parameters.AddWithValue("@psCBC", temp);
            cmd.Parameters.AddWithValue("@psIBID", internshipBookID);


            conn.Open();
            cmd.ExecuteNonQuery();
            conn.Close();
        }

        public static string getBookPage(int bookPageID)
        {
            MySqlConnection conn = new MySqlConnection(connString);

            string sql = "select pageText from bookpage where bookpageID=@pBPID";

            MySqlCommand cmd = new MySqlCommand(sql, conn);

            cmd.Parameters.AddWithValue("@pBPID", bookPageID);

            MySqlDataAdapter adaptor = new MySqlDataAdapter(cmd);

            DataSet dS = new DataSet();

            conn.Open();
            adaptor.Fill(dS);
            conn.Close();

            string pageText = "";

            if (dS.Tables[0].Rows.Count > 0)
            {
                pageText = dS.Tables[0].Rows[0][0].ToString();
            }

            return pageText;

        }

        public static DataSet getBookDate()
        {
            MySqlConnection conn = new MySqlConnection(connString);

            string sql = "select pageDate,bookpageID from bookpage where internshipBookID=@pIBID";

            MySqlCommand cmd = new MySqlCommand(sql, conn);

            cmd.Parameters.AddWithValue("@pIBID", internshipBookID);

            MySqlDataAdapter adaptor = new MySqlDataAdapter(cmd);

            DataSet dS = new DataSet();

            conn.Open();
            adaptor.Fill(dS);
            conn.Close();

            return dS;
        }

        public static void checkInternshipBookID()
        {
            MySqlConnection conn = new MySqlConnection(connString);

            string sql = "select internshipBookID from internshipbook where internshipID=@psCID and studentID=@psSID";

            MySqlCommand cmd = new MySqlCommand(sql, conn);

            cmd.Parameters.AddWithValue("@psSID", showInternshipAndBook.studentID);
            cmd.Parameters.AddWithValue("@psCID", showInternshipAndBook.internshipID);

            MySqlDataAdapter adaptor = new MySqlDataAdapter();
            adaptor.SelectCommand = cmd;

            DataSet dS = new DataSet();

            conn.Open();
            adaptor.Fill(dS);
            conn.Close();

            if (dS.Tables[0].Rows.Count > 0)
            {
                internshipBookID = Int32.Parse(dS.Tables[0].Rows[0][0].ToString());
            }
        }

        public static DataSet getInternshipInfo()
        {
            MySqlConnection conn = new MySqlConnection(connString);

            string sql = "select internship.internshipID,student.studentID,student.studentName,student.studentSurname,student.studentAverage,student.studentMail,student.studentProgram,student.studentClass,university.universityName,department.departmentName,university.universityName from internship inner join student on internship.studentID = student.studentID inner join university on university.universityID = student.universityID inner join department on department.departmentID = student.departmentID where internship.companyID=@psCID";

            MySqlCommand cmd = new MySqlCommand(sql, conn);

            cmd.Parameters.AddWithValue("@psCID", CompanyID);


            MySqlDataAdapter adaptor = new MySqlDataAdapter();
            adaptor.SelectCommand = cmd;

            DataSet dS = new DataSet();

            conn.Open();
            adaptor.Fill(dS);
            conn.Close();

            return dS;
        }


        public static DataSet getInternshipRequest()
        {
            MySqlConnection conn = new MySqlConnection(connString);

            string sql = "select student.studentID,student.studentName,student.studentSurname,student.studentAverage,student.studentMail,student.studentProgram,student.studentClass,department.departmentName,university.universityName from internshiprequestcompany inner join student on internshiprequestcompany.studentID = student.studentID inner join university on university.universityID = student.universityID inner join department on department.departmentID = student.departmentID where internshiprequestcompany.companyID=@psCID";

            MySqlCommand cmd = new MySqlCommand(sql, conn);

            cmd.Parameters.AddWithValue("@psCID", CompanyID);


            MySqlDataAdapter adaptor = new MySqlDataAdapter();
            adaptor.SelectCommand = cmd;

            DataSet dS = new DataSet();

            conn.Open();
            adaptor.Fill(dS);
            conn.Close();


            return dS;
        }

        public static void getCompanyMailAddress()
        {
            MySqlConnection conn = new MySqlConnection(connString);

            string sql = "select companyMail from company where companyID=@psCID";

            MySqlCommand cmd = new MySqlCommand(sql, conn);

            cmd.Parameters.AddWithValue("@psCID", CompanyID);


            MySqlDataAdapter adaptor = new MySqlDataAdapter();
            adaptor.SelectCommand = cmd;

            DataSet dS = new DataSet();

            conn.Open();
            adaptor.Fill(dS);
            conn.Close();

            try
            {
                companyMail = dS.Tables[0].Rows[0][0].ToString();
            }

            catch (IndexOutOfRangeException)
            {

            }
        }

        public static string checkInternshipRequestConfrim(int studentID)
        {
            MySqlConnection conn = new MySqlConnection(connString);

            string sql = "select companyRequestCheck from internshiprequestcompany where companyID=@pCID and studentID=@pSID";

            MySqlCommand cmd = new MySqlCommand(sql, conn);

            cmd.Parameters.AddWithValue("@pCID", CompanyID);
            cmd.Parameters.AddWithValue("@pSID", studentID);

            MySqlDataAdapter adaptor = new MySqlDataAdapter();
            adaptor.SelectCommand = cmd;

            DataSet dS = new DataSet();

            conn.Open();
            adaptor.Fill(dS);
            conn.Close();

            string temp = "";

            string result = "";

            if (dS.Tables[0].Rows.Count > 0)
            {
                temp = dS.Tables[0].Rows[0][0].ToString();

                if (temp == "confirm")
                {
                    result = "1";
                }
                else if (temp == "decline")
                {
                    result = "0";
                }
                else
                {
                    result = "2";
                }
            }
            else
            {
                result = "2";
            }

            return result;
        }

        public static void sendMailInternshipRequest(string mailText, string mailSubject, string mailDate, string studentMail)
        {
            MySqlConnection conn = new MySqlConnection(connString);

            string sql = "INSERT INTO mailbox(mailSubject,mailText,mailDate,senderMail,receiverMail) VALUES(@pMS,@pMT,@pMD,@pCM,@pSM)";

            MySqlCommand cmd = new MySqlCommand(sql, conn);

            cmd.Parameters.AddWithValue("@pMS", mailSubject);
            cmd.Parameters.AddWithValue("@pMT", mailText);
            cmd.Parameters.AddWithValue("@pMD", mailDate);
            cmd.Parameters.AddWithValue("@pCM", companyMail);
            cmd.Parameters.AddWithValue("@pSM", studentMail);

            conn.Open();
            cmd.ExecuteNonQuery();
            conn.Close();
        }

        public static void setInternshipRequest(string confirm, int studentID)
        {
            MySqlConnection conn = new MySqlConnection(connString);

            string sql = "update internshiprequestcompany set companyRequestCheck = @pcRC where studentID = @psID and companyID = @pCID";

            MySqlCommand cmd = new MySqlCommand(sql, conn);

            cmd.Parameters.AddWithValue("@pcRC", confirm);
            cmd.Parameters.AddWithValue("@psID", studentID);
            cmd.Parameters.AddWithValue("@pCID", CompanyID);

            conn.Open();
            cmd.ExecuteNonQuery();
            conn.Close();
        }
    }
}

File: Internship Tracking System - Student/src/internship/tracking/system/student/DBOperations.java
package internship.tracking.system.student;

import java.awt.HeadlessException;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;
import javafx.collections.FXCollections;
import javafx.collections.ObservableList;
import javax.swing.JOptionPane;

/**
 *
 * @author Arclus
 */

public class DBOperations
{
    Connection cn;
    
    static final ObservableList cOptions = FXCollections.observableArrayList();
    static final ObservableList dOptions = FXCollections.observableArrayList();
    static final ObservableList studentInfoList = FXCollections.observableArrayList();
    static final ObservableList internshipRequestUniversity = FXCollections.observableArrayList();
    static final ObservableList iRUC = FXCollections.observableArrayList();
    static final ObservableList checkInternship = FXCollections.observableArrayList();
    static final ObservableList gIInfo = FXCollections.observableArrayList();
    static final ObservableList gMInbox = FXCollections.observableArrayList();
    public static int universityID;
    public static int internshipBookID;
    public static String mail;
    
    
    public Statement connection()
    {
        try
        {
            Class.forName("com.mysql.jdbc.Driver").newInstance();
            cn = DriverManager.getConnection("jdbc:mysql://localhost:3306/interndb?autoReconnect=true&useSSL=false", "username", "password");
            
            return cn.createStatement();
        }
        
        catch(ClassNotFoundException | InstantiationException | IllegalAccessException | SQLException | HeadlessException e)
        {
            JOptionPane.showMessageDialog(null, "Not Connected");
            
            return null;
        }
    }
    
    public static String loginCheck(String id, String pW)
    {
        DBOperations vt = new DBOperations();
        Statement connection = vt.connection();
        
        int tempID = Integer.parseInt(id);
        
        try
        {
            String sql= "SELECT studentID,studentPassword,studentMail FROM student WHERE studentID = '" + tempID + "' && studentPassword = '" + pW + "'";
            ResultSet rss = connection.executeQuery(sql);
            
            if (rss.next())
            {
                InternshipTrackingSystemStudent.studentMail = rss.getString("studentMail");
                rss.close();
                connection.close();
                
                return "1";
            }

            else
            {
                rss.close();
                connection.close();
                
                return "0";
            }
        }
        catch (SQLException | HeadlessException a)
        {
           return a.toString();
        }
    }
    
    public static String checkUniversityID(String universityEmail)
    {
        DBOperations vt = new DBOperations();
        Statement connection = vt.connection();
        
        try
        {
            String sql= "SELECT universityID FROM university WHERE universityEmail = '" + universityEmail + "'";
            
            try (ResultSet rss = connection.executeQuery(sql))
            {  
                while(rss.next())
                {  
                    System.out.println("girdi mi");
                    return rss.getString("universityID");
                }
                
                connection.close();
                rss.close();
                
                return "0";
            }     
        }
        catch (SQLException | HeadlessException a)
        {
           return a.toString();
        }
    }
    
        public static String checkCompanyID(String companyEmail)
    {
        DBOperations vt = new DBOperations();
        Statement connection = vt.connection();
        
        try
        {
            String sql= "SELECT companyID FROM company WHERE companyMail = '" + companyEmail + "'";
            
            try (ResultSet rss = connection.executeQuery(sql))
            {  
                while(rss.next())
                {  
                    
                    return rss.getString("companyID");
                }
                
                connection.close();
                rss.close();
                
                return "0";
            }     
        }
        catch (SQLException | HeadlessException a)
        {
           return a.toString();
        }
    }
        
    
    public static String getMailOutbox()
    {
        DBOperations vt = new DBOperations();
        Statement connection = vt.connection();
        
        try
        {
            String sql= "SELECT mailID,receiverMail,mailSubject,mailText,mailDate FROM mailbox WHERE senderMail = '" + InternshipTrackingSystemStudent.studentMail + "'";
            
            try (ResultSet rss = connection.executeQuery(sql))
            {       
                while (rss.next())
                {
                     MailBoxPage.data.add(new MailTable(
                        rss.getString("mailID"),
                        rss.getString("receiverMail"),
                        rss.getString("mailSubject"),
                        rss.getString("mailText"),
                        rss.getString("mailDate")
                    ));
                }
                MailBoxPage.table.setItems(MailBoxPage.data);
                
                connection.close();
                rss.close();
            }
            
            return "1";     
        }
        catch (SQLException | HeadlessException a)
        {
           return a.toString();
        }
    }
                
    public static String getMailInbox()
    {
        DBOperations vt = new DBOperations();
        Statement connection = vt.connection();
        
        try
        {
            String sql= "SELECT mailID,senderMail,mailSubject,mailText,mailDate FROM mailbox WHERE receiverMail = '" + InternshipTrackingSystemStudent.studentMail + "'";
            
            try (ResultSet rss = connection.executeQuery(sql))
            {       
                while (rss.next())
                {
                     MailBoxPage.data.add(new MailTable(
                        rss.getString("mailID"),
                        rss.getString("senderMail"),
                        rss.getString("mailSubject"),
                        rss.getString("mailText"),
                        rss.getString("mailDate")
                    ));
                }
                MailBoxPage.table.setItems(MailBoxPage.data);
                
                connection.close();
                rss.close();
            }
            
            return "1";     
        }
        catch (SQLException | HeadlessException a)
        {
           return a.toString();
        }
    }
        
    public static String sendMailCompany(String companyMail, String mailSubject, String mailText, String mailDate)
    {
        DBOperations vt = new DBOperations();
        Statement connection = vt.connection();
        
        try
        {
            String sql= "INSERT INTO mailbox(mailSubject,mailText,mailDate,senderMail,receiverMail) VALUES (" + " '" + mailSubject + "', " + " '" + mailText + "', " + " '" + mailDate + "', " + " '" + InternshipTrackingSystemStudent.studentMail + "', " + " '" + companyMail + "')";
            
            connection.executeUpdate(sql);
            connection.close();
            
            return "1";
        }
        catch (SQLException | HeadlessException a)
        {
           return a.toString();
        }
    }
    
    public static String sendMailUniversity(String universityMail, String mailSubject, String mailText, String mailDate)
    {
        DBOperations vt = new DBOperations();
        Statement connection = vt.connection();
        
        try
        {
            String sql= "INSERT INTO mailbox(mailSubject,mailText,mailDate,senderMail,receiverMail) VALUES (" + " '" + mailSubject + "', " + " '" + mailText + "', " + " '" + mailDate + "', " + " '" + InternshipTrackingSystemStudent.studentMail + "', " + " '" + universityMail + "')";
            
            connection.executeUpdate(sql);
            connection.close();
            
            return "1";
        }
        catch (SQLException | HeadlessException a)
        {
           return a.toString();
        }
    }
    
    public static String insertPDF(String fileName)
    {
        DBOperations vt = new DBOperations();
        Statement connection = vt.connection();
        
        String sql= "SELECT internship.internshipID FROM internship";

        return "0";
    }
    
    public static String addInternshipBookPage(String pageText, String date)
    {
        DBOperations vt = new DBOperations();
        Statement connection = vt.connection();
        
        try
        {
            String sql= "INSERT INTO bookpage(pageText,internshipBookID,pageDate) VALUES (" + " '" + pageText + "', " + " '" + internshipBookID + "', " + " '" + date + "')";
            
            connection.executeUpdate(sql);
            connection.close();
            
            return "1";
        }
        catch (SQLException | HeadlessException a)
        {
           return a.toString();
        }
    }
    
    public static String checkInternship(int id)
    {
        DBOperations vt = new DBOperations();
        Statement connection = vt.connection();
        
        try
        {
            String sql= "SELECT internship.internshipID FROM internship INNER JOIN student on internship.internshipID = student.studentID WHERE internship.studentID = '" + id + "'";
            
            try (ResultSet rss = connection.executeQuery(sql))
            {  
                while(rss.next())
                {   
                    checkInternship.add(rss.getString("internship.internshipID"));
                    return "1";
                }
                
                connection.close();
                rss.close();
                
                return "0";
            }     
        }
        catch (SQLException | HeadlessException a)
        {
           return a.toString();
        }
    }
    
    public static String getInternshipInfo(int comboID)
    {
        DBOperations vt = new DBOperations();
        Statement connection = vt.connection();
        
        try
        {
            String sql= "SELECT internship.internshipID,internship.program,company.companyName,internship.startDate,internship.finishDate,internship.acceptedDay,internship.universtiyCheck,internship.companyCheck  FROM internship INNER JOIN company on company.companyID = internship.companyID WHERE internshipID = '" + comboID + "'";
            
            try (ResultSet rss = connection.executeQuery(sql))
            {  
                while(rss.next())
                {   
                    gIInfo.add(rss.getString("internship.internshipID"));
                    gIInfo.add(rss.getString("internship.program"));
                    gIInfo.add(rss.getString("company.companyName"));
                    gIInfo.add(rss.getString("internship.startDate"));
                    gIInfo.add(rss.getString("internship.finishDate"));
                    gIInfo.add(rss.getString("internship.acceptedDay"));
                    gIInfo.add(rss.getString("internship.universtiyCheck"));
                    gIInfo.add(rss.getString("internship.companyCheck"));
                }
                
                connection.close();
                rss.close();
            }
            
            return "1";
        }
        catch (SQLException | HeadlessException a)
        {
           return a.toString();
        }
    }
    
    public static String getStudentInfo(int id)
    {
        DBOperations vt = new DBOperations();
        Statement connection = vt.connection();
       
        try
        {
            String sql= "SELECT student.studentName,student.studentSurname,"
                    + "student.studentID,student.studentClass,"
                    + "student.studentAverage,department.departmentName,student.universityID "
                    + "FROM student "
                    + "INNER JOIN department on department.departmentID = student.departmentID "
                    + "WHERE studentID = '" + id + "'";
            
            try (ResultSet rss = connection.executeQuery(sql))
            {  
                while(rss.next())
                {   
                    studentInfoList.add(rss.getString("student.studentName"));
                    studentInfoList.add(rss.getString("student.studentSurname"));
                    studentInfoList.add(rss.getString("student.studentID"));
                    studentInfoList.add(rss.getString("student.studentClass"));
                    studentInfoList.add(rss.getString("student.studentAverage"));
                    studentInfoList.add(rss.getString("department.departmentName"));
                    universityID = Integer.parseInt(rss.getString("student.universityID"));
                }
                
                connection.close();
                rss.close();
            }
            
            return "1";
        }
        catch (SQLException | HeadlessException a)
        {
           return a.toString();
        }
    }
    
    public static String requestInternUniversity(int studentID, int universityID, int internshipTerm, int internshipDay, int iDWeek, String iStartDate, String iEndDate, String program)
    {
        DBOperations vt = new DBOperations();
        Statement connection = vt.connection();
        
        try
        {
            String sql= "INSERT INTO internshiprequestuniversity(studentID,universityID,universityProgram,internshipTerm,internshipDay,internshipDayWeek,internshipStartDate,internshipEndDate) VALUES (" + " '" + studentID + "', " + " '" + universityID + "', " + " '" + program + "', " + " '" + internshipTerm + "', " + " '" + internshipDay + "', " + " '" + iDWeek + "', " + " '" + iStartDate + "', " + " '" + iEndDate + "')";
            
            connection.executeUpdate(sql);
            connection.close();
            
            return "1";
        }
        catch (SQLException | HeadlessException a)
        {
           return a.toString();
        }
    }
    
    public static String requestInternCompany(int studentID, int selectedCompany)
    {
        DBOperations vt = new DBOperations();
        Statement connection = vt.connection();
        
        try
        {
            String sql= "INSERT INTO internshiprequestcompany(studentID,companyID) VALUES (" + " '" + studentID + "', " + " '" + selectedCompany + "')";
            
            connection.executeUpdate(sql);
            connection.close();
            
            return "1";
        }
        catch (SQLException | HeadlessException a)
        {
           return a.toString();
        }
    }
    
    public static String getMailInbox(int id)
    {
        DBOperations vt = new DBOperations();
        Statement connection = vt.connection();
        
        try
        {
            String sql= "SELECT senderID,mailSubject,mailText,mailDate FROM mailbox WHERE receiverID = '" + id + "'";
            
            try (ResultSet rss = connection.executeQuery(sql))
            {  
                while(rss.next())
                {   
                    gMInbox.add(rss.getString("mailSubject"));
                    gMInbox.add(rss.getString("mailText"));
                    gMInbox.add(rss.getString("mailDate"));
                    gMInbox.add(rss.getString("senderID"));
                    
                    return "1";
                }
                
                connection.close();
                rss.close();
                
                return "0";
            }     
        }
        catch (SQLException | HeadlessException a)
        {
           return a.toString();
        }
    }
    
    public static String getRequestInternInfo(int id)
    {
        DBOperations vt = new DBOperations();
        Statement connection = vt.connection();
        
        try
        {
            String sql= "SELECT company.companyName,company.companyCity,"
                    + "company.companyFieldActivity,company.companyMail,"
                    + "company.companyPhone,company.companyAddress "
                    + "FROM internshiprequestcompany "
                    + "INNER JOIN student on internshiprequestcompany.studentID = student.studentID "
                    + "INNER JOIN company on company.companyID = internshiprequestcompany.companyID "
                    + "WHERE student.studentID = '" + id + "'";
            
            try (ResultSet rss = connection.executeQuery(sql))
            {
                while(rss.next())
                {
                    internshipRequestUniversity.add(rss.getString("company.companyName"));
                    iRUC.add(rss.getString("company.companyCity"));
                    internshipRequestUniversity.add(rss.getString("company.companyFieldActivity"));
                    internshipRequestUniversity.add(rss.getString("company.companyMail"));
                    internshipRequestUniversity.add(rss.getString("company.companyPhone"));
                    internshipRequestUniversity.add(rss.getString("company.companyAddress"));
                }
                
                connection.close();
                rss.close();
            }
            
            return "1";
        }
        catch (SQLException | HeadlessException a)
        {
           return a.toString();
        }
    }
    
    public static String checkCompanyConfirm(int id)
    {
        DBOperations vt = new DBOperations();
        Statement connection = vt.connection();
        
        String temp = "1";
        
        try
        {
            String sql= "SELECT internshiprequestcompany.companyRequestCheck"
                    + " FROM internshiprequestcompany "
                    + "INNER JOIN student on internshiprequestcompany.studentID = student.studentID "
                    + "WHERE student.studentID = '" + id + "'";
            
            try (ResultSet rss = connection.executeQuery(sql))
            {  
                while(rss.next())
                {   
                    temp = rss.getString("internshiprequestcompany.companyRequestCheck");
                }
                
                connection.close();
                rss.close();
            }
            
            return temp;
        }
        catch (SQLException | HeadlessException a)
        {
           return a.toString();
        }
    }
    
    public static String checkInternshipBookID()
    {
        DBOperations vt = new DBOperations();
        Statement connection = vt.connection();
        
        try
        {
            String sql= "SELECT internshipBookID FROM internshipbook "
                    + "WHERE internshipID = '" + ShowIntern.selectedInternship + "' "
                    + "AND studentID = '" + InternshipTrackingSystemStudent.studentNumber + "'";
            
            try (ResultSet rss = connection.executeQuery(sql))
            {  
                while(rss.next())
                {
                    internshipBookID = Integer.parseInt(rss.getString("internshipBookID"));
                    return "1";
                }
                
                connection.close();
                rss.close();
                
                return "0";
            }     
        }
        catch (SQLException | HeadlessException a)
        {
           return a.toString();
        }
    }
    
    public static String deleteInternshipBookPage(String pageDate)
    {
        DBOperations vt = new DBOperations();
        Statement connection = vt.connection();
        
        try
        {
            String sql= "DELETE FROM bookpage WHERE pageDate = '" + pageDate + "' AND internshipBookID = '" + internshipBookID + "'";
                   
            connection.executeUpdate(sql);
            connection.close();
            return "1";
        }
        catch (SQLException | HeadlessException a)
        {
            return "a"; 
        }
    }
    
    public static String updateInternshipBookPage(String pageText, String pageDate)
    {
        DBOperations vt = new DBOperations();
        Statement connection = vt.connection();
        
        try
        {
            String sql= "UPDATE bookpage SET pageText = '" + pageText + "' WHERE internshipBookID = '" + internshipBookID + "' AND pageDate = '" + pageDate + "'";
            
            connection.executeUpdate(sql);
            connection.close();
            
            return "1";
        }
        catch (SQLException | HeadlessException a)
        {
           return a.toString();
        }
    }
    
    public static String getMailForUpdate(String pageDate)
    {
        DBOperations vt = new DBOperations();
        Statement connection = vt.connection();
        
        try
        {
            String sql= "SELECT pageText FROM bookpage WHERE internshipBookID = '" + internshipBookID + "' AND pageDate = '" + pageDate + "'";

            try (ResultSet rss = connection.executeQuery(sql))
            {       
                while (rss.next())
                {
                    mail = rss.getString("pageText");
                    
                    return "1"; 
                }
                connection.close();
                rss.close();
            }
            
            return "0";     
        }
        
        catch (SQLException | HeadlessException a)
        {
           return a.toString();
        }
    }
    
    public static String getDate()
    {
        DBOperations vt = new DBOperations();
        Statement connection = vt.connection();
 
        String sql= "SELECT pageDate FROM bookpage WHERE internshipBookID = '" + internshipBookID + "'";

        try (ResultSet rss = connection.executeQuery(sql))
        {       
            while (rss.next())
            {
                dOptions.add(rss.getString("pageDate"));
            }
            
            connection.close();
            rss.close();

            return "1";     
        }
        
        catch (SQLException | HeadlessException a)
        {
           return a.toString();
        }
    }
    
    public static String getCity()
    {
        DBOperations vt = new DBOperations();
        Statement connection = vt.connection();
        
        try
        {
            String sql= "SELECT cityName FROM city";
            
            try (ResultSet rss = connection.executeQuery(sql))
            {       
                while (rss.next())
                {
                    cOptions.add(rss.getString("cityName"));
                }
                connection.close();
                rss.close();
            }
            
            return "1";     
        }
        
        catch (SQLException | HeadlessException a)
        {
           return a.toString();
        }
    }
    
    public static String getCompany(String selectedcityName)
    {
        DBOperations vt = new DBOperations();
        Statement connection = vt.connection();
        
        try
        {
            String sql= "SELECT companyID,companyName,companyCity,companyFieldActivity,companyPhone,companyMail,companyAddress FROM company WHERE companyCity = '" + selectedcityName + "'";
            
            try (ResultSet rss = connection.executeQuery(sql))
            {       
                while (rss.next())
                {
                     RequestCompany.data.add(new CompanyTable(
                        rss.getString("companyID"),
                        rss.getString("companyName"),
                        rss.getString("companyCity"),
                        rss.getString("companyFieldActivity"),
                        rss.getString("companyPhone"),
                        rss.getString("companyMail"),
                        rss.getString("companyAddress")
                    ));
                }
                RequestCompany.table.setItems(RequestCompany.data);
                
                connection.close();
                rss.close();
            }
            
            return "1";     
        }
        catch (SQLException | HeadlessException a)
        {
           return a.toString();
        }
    }
}